spec.html

```diff
@@ -756,5 +756,76 @@ <h1>%AsyncIterator.prototype% [ @@toStringTag ]</h1>
         <p>This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</p>
       </emu-clause>
     </emu-clause>
+
+    <emu-clause id="sec-%asyncfromsynciteratorprototype%-object">
+      <h1>The %AsyncFromSyncIteratorPrototype% Object</h1>
+
+      <emu-clause id="sec-%asyncfromsynciteratorprototype%.next">
+        <h1>%AsyncFromSyncIteratorPrototype%.next ( _value_ )</h1>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. <del>Assert: Type(_O_) is Object and _O_ has a [[SyncIteratorRecord]] internal slot.</del>
+          1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
+          1. <ins>If Type(_O_) is not Object, or if _O_ does not have a [[SyncIteratorRecord]] internal slot, then</ins>
```

Comment:
you can use RequireInternalSlot instead.

Reply:
That will cause the function to throw a `TypeError` instead of returning a rejected `Promise`.

Reply:
IfAbruptRejectPromise may be useful here